From: "David S. Miller" <davem@davemloft.net>
Date: Tue, 6 Jun 2017 11:34:06 -0400
Subject: Revert "sit: reload iphdr in ipip6_rcv"
Origin: https://git.kernel.org/linus/f4eb17e1efe538d4da7d574bedb00a8dafcc26b7
Bug-Debian: https://bugs.debian.org/903776

This reverts commit b699d0035836f6712917a41e7ae58d84359b8ff9.

As per Eric Dumazet, the pskb_may_pull() is a NOP in this
particular case, so the 'iph' reload is unnecessary.

Signed-off-by: David S. Miller <davem@davemloft.net>
[carnil: cherry pick 5b8fcc075714b86fb8fe194bb463fed2998a8e85 which is
the backport of f4eb17e1efe538d4da7d574bedb00a8dafcc26b7 in 4.9.112] 
---
 net/ipv6/sit.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/net/ipv6/sit.c b/net/ipv6/sit.c
index ae0485d776f4..fc7ca1e46908 100644
--- a/net/ipv6/sit.c
+++ b/net/ipv6/sit.c
@@ -659,7 +659,6 @@ static int ipip6_rcv(struct sk_buff *skb)
 		if (iptunnel_pull_header(skb, 0, htons(ETH_P_IPV6),
 		    !net_eq(tunnel->net, dev_net(tunnel->dev))))
 			goto out;
-		iph = ip_hdr(skb);
 
 		err = IP_ECN_decapsulate(iph, skb);
 		if (unlikely(err)) {
-- 
2.18.0

